From 2eef57ab20128edd8432e7857d9d726463301ab3 Mon Sep 17 00:00:00 2001
From: Joe Honold <mozzwald@gmail.com>
Date: Thu, 14 Jul 2016 18:18:17 -0500
Subject: [PATCH] power_supply: use oneshot inverted for
 charging_blink_full_solid (full) Use oneshot inverted blink so
 charging_blink_full_solid works as intended on FULL status. Using
 led_trigger_event(psy->charging_blink_full_solid_trig, LED_FULL) actually
 turns off the LED, possibly a bug in core led code?

Signed-off-by: Joe Honold <mozzwald@gmail.com>
---
 drivers/power/power_supply_leds.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_leds.c b/drivers/power/power_supply_leds.c
index 2277ad9..5a78509 100644
--- a/drivers/power/power_supply_leds.c
+++ b/drivers/power/power_supply_leds.c
@@ -35,8 +35,13 @@ static void power_supply_update_bat_leds(struct power_supply *psy)
 		led_trigger_event(psy->charging_full_trig, LED_FULL);
 		led_trigger_event(psy->charging_trig, LED_OFF);
 		led_trigger_event(psy->full_trig, LED_FULL);
-		led_trigger_event(psy->charging_blink_full_solid_trig,
-			LED_FULL);
+		/*
+		 * do oneshot inverted blink so charging_blink_full_solid works
+		 * as intended. Using led_trigger_event actually turns off the
+		 * LED, possibly a bug in core led code?
+		*/
+		led_trigger_blink_oneshot(psy->charging_blink_full_solid_trig,
+			&delay_on, &delay_off, 1);
 		break;
 	case POWER_SUPPLY_STATUS_CHARGING:
 		led_trigger_event(psy->charging_full_trig, LED_FULL);
-- 
2.1.4

